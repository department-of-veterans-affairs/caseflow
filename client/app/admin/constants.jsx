export const ACTIONS = {
  SET_FEATURE_TOGGLES: 'SET_FEATURE_TOGGLES',
  POST_VETERAN_EXTRACT: 'POST_VETERAN_EXTRACT',
  STARTED_VETERAN_EXTRACT: 'STARTED_VETERAN_EXTRACT',
  FAILED_VETERAN_EXTRACT: 'FAILED_VETERAN_EXTRACT'
};
