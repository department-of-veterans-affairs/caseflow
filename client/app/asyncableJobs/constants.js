export const ACTIONS = {
  MANUAL_JOB_STARTED: 'MANUAL_JOB_STARTED',
  MANUAL_JOB_FAILED: 'MANUAL_JOB_FAILED',
  CLEAR_MANUAL_JOB_TYPE: 'CLEAR_MANUAL_JOB_TYPE',
};