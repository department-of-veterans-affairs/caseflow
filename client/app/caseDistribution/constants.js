import DISTRIBUTION from '../../constants/DISTRIBUTION';

const LEVERS = 'levers';

export const Constant = {
  LEVERS
};

export const sectionTitles = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]: DISTRIBUTION.ama_hearings_section_title,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]: DISTRIBUTION.ama_direct_review_section_title,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]: DISTRIBUTION.ama_evidence_submission_section_title
};

export const docketTimeGoalPriorMappings = {
  [DISTRIBUTION.ama_hearings_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_hearings_docket_time_goals,
  [DISTRIBUTION.ama_direct_review_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_direct_review_docket_time_goals,
  [DISTRIBUTION.ama_evidence_submission_start_distribution_prior_to_goals]:
    DISTRIBUTION.ama_evidence_submission_docket_time_goals
};
