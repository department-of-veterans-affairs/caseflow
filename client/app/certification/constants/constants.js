// actions
//
export const UPDATE_PROGRESS_BAR = 'UPDATE_PROGRESS_BAR';
export const TOGGLE_CANCELLATION_MODAL = 'TOGGLE_CANCELLATION_MODAL';
export const CHANGE_VBMS_HEARING_DOCUMENT = 'CHANGE_VBMS_HEARING_DOCUMENT';
export const CHANGE_TYPE_OF_FORM9 = 'CHANGE_TYPE_OF_FORM9';
export const CHANGE_TYPE_OF_HEARING = 'CHANGE_TYPE_OF_HEARING';

export const CHANGE_REPRESENTATIVE_NAME = 'CHANGE_REPRESENTATIVE_NAME';
export const CHANGE_REPRESENTATIVE_TYPE = 'CHANGE_REPRESENTATIVE_TYPE';
export const CHANGE_OTHER_REPRESENTATIVE_TYPE = 'CHANGE_OTHER_REPRESENTATIVE_TYPE';

export const CHANGE_SIGN_AND_CERTIFY_FORM = 'CHANGE_SIGN_AND_CERTIFY_FORM';

export const CERTIFICATION_UPDATE_REQUEST = 'CERTIFICATION_UPDATE_REQUEST';
export const CERTIFICATION_UPDATE_FAILURE = 'CERTIFICATION_UPDATE_FAILURE';
export const CERTIFICATION_UPDATE_SUCCESS = 'CERTIFICATION_UPDATE_SUCCESS';

export const ON_CONTINUE_CLICK_FAILED = 'ON_CONTINUE_CLICK_FAILED';
export const ON_CONTINUE_CLICK_SUCCESS = 'ON_CONTINUE_CLICK_SUCCESS';

export const RESET_STATE = 'RESET_STATE';

// types of hearings
//
// TODO:
// on the backend, NO_HEARING_DESIRED
// should result in VIDEO being written to VACOLS,
// and HEARING_CANCELLED should result in a cancellation
// checkbox being checked, but the original hearing type
// should be undisturbed.
export const hearingPreferences = {
  VIDEO: 'VIDEO',
  TRAVEL_BOARD: 'TRAVEL_BOARD',
  WASHINGTON_DC: 'WASHINGTON_DC',
  HEARING_TYPE_NOT_SPECIFIED: 'HEARING_TYPE_NOT_SPECIFIED',
  NO_HEARING_DESIRED: 'NO_HEARING_DESIRED',
  HEARING_CANCELLED: 'HEARING_CANCELLED',
  NO_BOX_SELECTED: 'NO_BOX_SELECTED'
};

// form9 values
export const form9Types = {
  FORMAL_FORM9: 'FORMAL_FORM9',
  INFORMAL_FORM9: 'INFORMAL_FORM9'
};

// representation for the appellant
export const representativeTypes = {
  ATTORNEY: 'ATTORNEY',
  AGENT: 'AGENT',
  ORGANIZATION: 'ORGANIZATION',
  NONE: 'NONE',
  // TODO: should "Other be a real type"?
  OTHER: 'OTHER'
};

// was a hearing document found in VBMS?
export const vbmsHearingDocument = {
  FOUND: 'FOUND',
  NOT_FOUND: 'NOT_FOUND'
};

export const progressBarSections = {
  CHECK_DOCUMENTS: 'CHECK_DOCUMENTS',
  CONFIRM_CASE_DETAILS: 'CONFIRM_CASE_DETAILS',
  CONFIRM_HEARING: 'CONFIRM_HEARING',
  SIGN_AND_CERTIFY: 'SIGN_AND_CERTIFY'
};

export const certifyingOfficialTitles = {
  DECISION_REVIEW_OFFICER: 'DECISION_REVIEW_OFFICER',
  RATING_SPECIALIST: 'RATING_SPECIALIST',
  VETERANS_SERVICE_REPRESENTATIVE: 'VETERANS_SERVICE_REPRESENTATIVE',
  CLAIMS_ASSISTANT: 'CLAIMS_ASSISTANT',
  OTHER: 'OTHER'
};
