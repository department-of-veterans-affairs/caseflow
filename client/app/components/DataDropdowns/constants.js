import _ from 'lodash';

import HEARING_ROOMS_LIST from '../../../constants/HEARING_ROOMS_LIST.json';

export const HEARING_ROOM_OPTIONS = _.map(HEARING_ROOMS_LIST, (value, key) => ({
  label: value.label,
  value: key.toString()
}));
