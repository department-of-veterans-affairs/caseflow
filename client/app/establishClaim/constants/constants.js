// actions
export const CHANGE_SPECIAL_ISSUE = 'CHANGE_SPECIAL_ISSUE';
