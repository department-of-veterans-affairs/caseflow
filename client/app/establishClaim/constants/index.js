// actions
export const CHANGE_SPECIAL_ISSUE = 'CHANGE_SPECIAL_ISSUE';
export const CHANGE_ESTABLISH_CLAIM_FIELD = 'CHANGE_ESTABLISH_CLAIM_FIELD';
export const TOGGLE_CANCEL_TASK_MODAL = 'TOGGLE_CANCEL_TASK_MODAL';
export const TRIGGER_LOADING = 'TRIGGER_LOADING';
export const CHANGE_CANCEL_FEEDBACK = 'CHANGE_CANCEL_FEEDBACK';
export const VALIDATE_CANCEL_FEEDBACK = 'VALIDATE_CANCEL_FEEDBACK';
export const REQUEST_CANCEL_FEEDBACK = 'REQUEST_CANCEL_FEEDBACK';
export const REQUEST_CANCEL_FEEDBACK_FAILURE = 'REQUEST_CANCEL_FEEDBACK_FAILURE';
export const INCREMENT_MODIFIER_ON_DUPLICATE_EP_ERROR = 'INCREMENT_MODIFIER_ON_DUPLICATE_EP_ERROR';

// constants
export const FULL_GRANT = 'Full Grant';
export const MODIFIER_OPTIONS = [
  '070',
  '071',
  '072',
  '073',
  '074',
  '075',
  '076',
  '077',
  '078',
  '079'
];

export const ARC_STATION_OF_JURISDICTION = '397';
