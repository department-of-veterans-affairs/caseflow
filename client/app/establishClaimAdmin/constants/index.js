export const SET_LOADING_STATE = 'SET_LOADING_STATE';
export const POPULATE_STUCK_TASKS = 'POPULATE_STUCK_TASKS';
