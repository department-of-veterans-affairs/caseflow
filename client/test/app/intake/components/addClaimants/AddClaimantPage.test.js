import React from 'react';
import { screen, waitFor } from '@testing-library/react';
import { axe } from 'jest-axe';
import userEvent from '@testing-library/user-event';
import selectEvent from 'react-select-event';
import { createMemoryHistory } from 'history';

import { AddClaimantPage } from 'app/intake/addClaimant/AddClaimantPage';
import { renderIntakePage } from '../../testUtils';
import { generateInitialState } from 'app/intake/index';
import { PAGE_PATHS } from 'app/intake/constants';
import { fillForm, relationshipOpts } from './testUtils';

describe('AddClaimantPage', () => {
  const onSubmit = jest.fn();
  const onBack = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });
  const defaults = { onSubmit, onBack };
  const setup = (
    storeValues,
    history = createMemoryHistory({ initialEntries: [PAGE_PATHS.ADD_CLAIMANT] }),
  ) => {
    const page = <AddClaimantPage {...defaults} />;

    return renderIntakePage(page, storeValues, history);
  };

  it('renders default state correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
    expect(screen.getByText('Add Claimant')).toBeInTheDocument();
  });

  it('passes a11y testing', async () => {
    const { container } = setup();

    const results = await axe(container);

    expect(results).toHaveNoViolations();
  });

  it('fires onBack', async () => {
    setup();

    const backButton = screen.getByRole('button', { name: /back/i });

    expect(onBack).not.toHaveBeenCalled();

    await userEvent.click(backButton);
    await waitFor(() => {
      expect(onBack).not.toHaveBeenCalled();
    });
  });

  describe('form validation', () => {
    it('disables submit button unless valid', async () => {
      setup();

      const submit = screen.getByRole('button', {
        name: /Continue to next step/i,
      });

      // submit button disabled
      await waitFor(() => {
        expect(submit).toBeDisabled();
      });

      // Select option
      await selectEvent.select(
        screen.getByLabelText('Relationship to the Veteran'),
        [relationshipOpts[3].label]
      );

      // Set organization
      await userEvent.click(
        screen.getByRole('radio', { name: /organization/i })
      );

      // Wait for form to re-render
      await waitFor(() => {
        expect(
          screen.getByRole('textbox', { name: /Organization name/i })
        ).toBeInTheDocument();
      });

      await fillForm();
      // trigger onBlur
      userEvent.tab();
      // submit button enabled
      await waitFor(() => {
        expect(submit).not.toBeDisabled();
      });
    }, 15000);
  });

  describe('Redirection to Intake home page', () => {
    let storeValues;

    beforeEach(() => {
      storeValues = generateInitialState();
    });

    it('takes place whenever intake has been cancelled (formType === null)', async () => {
      storeValues.intake = {
        ...storeValues.intake,
        formType: null
      };

      const { history } = setup(storeValues);

      expect(await history.location.pathname).toBe(PAGE_PATHS.BEGIN);
    });

    it('does not take place is there is a formType, indicating no cancellation', async () => {
      storeValues.intake = {
        ...storeValues.intake,
        formType: 'appeal'
      };

      const { history } = setup(storeValues);

      expect(await history.location.pathname).toBe(PAGE_PATHS.ADD_CLAIMANT);
    });
  });
});
