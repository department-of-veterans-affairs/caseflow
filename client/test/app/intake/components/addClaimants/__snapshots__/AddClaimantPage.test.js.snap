// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddClaimantPage renders default state correctly 1`] = `
<div>
  <div
    class="cf-app-segment"
  >
    <div
      class="cf-progress-bar"
      role="list"
    >
      <div
        aria-current="step"
        class="cf-progress-bar-section"
        role="listitem"
      >
        <div
          class="cf-progress-bar-activated"
        >
          <b>
            1. Select Form
          </b>
        </div>
      </div>
      <div
        aria-current="false"
        class="cf-progress-bar-section"
        role="listitem"
      >
        <div
          class="cf-progress-bar-not-activated"
        >
          2. Search
        </div>
      </div>
      <div
        aria-current="false"
        class="cf-progress-bar-section"
        role="listitem"
      >
        <div
          class="cf-progress-bar-not-activated"
        >
          3. Review
        </div>
      </div>
      <div
        aria-current="false"
        class="cf-progress-bar-section"
        role="listitem"
      >
        <div
          class="cf-progress-bar-not-activated"
        >
          4. Add Issues
        </div>
      </div>
      <div
        aria-current="false"
        class="cf-progress-bar-section"
        role="listitem"
      >
        <div
          class="cf-progress-bar-not-activated"
        >
          5. Confirmation
        </div>
      </div>
    </div>
  </div>
  <section
    class="cf-app-segment cf-app-segment--alt"
  >
    <div>
      <h1>
        Add Claimant
      </h1>
      <p>
        You are adding a claimant who is not listed in Caseflow. Please select the claimantâ€™s relationship to the Veteran and add their information accordingly. If the claimant has indicated a representative, you can add their information in the next step.
      </p>
      <form>
        <div
          class=""
        >
          <div
            class="cf-form-dropdown dropdown-relationship"
          >
            <label
              class="question-label"
              for="relationship"
              id="relationship-label"
            >
              <strong>
                <span>
                  Relationship to the Veteran
                </span>
              </strong>
            </label>
            <div
              class="cf-select"
            >
              <div
                class=" css-2b097c-container"
              >
                <div
                  class="cf-select__control css-yk16xz-control"
                >
                  <div
                    class="cf-select__value-container css-g1d714-ValueContainer"
                  >
                    <div
                      class="cf-select__placeholder css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      class="css-81nyic-Input"
                    >
                      <div
                        class="cf-select__input"
                        style="display: inline-block;"
                      >
                        <input
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-labelledby="relationship-label"
                          aria-owns="relationship-listbox"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          id="relationship"
                          role="combobox"
                          spellcheck="false"
                          style="box-sizing: content-box; width: 2px; border: 0px; opacity: 1; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="cf-select__indicators css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      class="cf-select__indicator-separator css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      class="cf-select__indicator cf-select__dropdown-indicator css-tlfecz-indicatorContainer"
                    >
                      <svg
                        aria-hidden="true"
                        class="css-6q0nyr-Svg"
                        focusable="false"
                        height="20"
                        viewBox="0 0 20 20"
                        width="20"
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
                <input
                  name="relationship"
                  type="hidden"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <br />
        <br />
      </form>
    </div>
  </section>
  <nav
    class="cf-app-segment css-syjz65"
    role="navigation"
  >
    <div>
      <button
        class="cf-btn-link cf-btn-link usa-button"
        id="cancel-intake"
        style="float: left; padding-left: 0px;"
        type="button"
      >
        Cancel intake
      </button>
      <span>
        <button
          class="usa-button-secondary usa-button"
          id="button-go-back"
          style="margin-right: 1em;"
          type="button"
        >
          Back
        </button>
      </span>
      <span>
        <button
          class="cf-right-side usa-button"
          id="button-submit-review"
          type="button"
        >
          Continue to next step
        </button>
      </span>
    </div>
  </nav>
</div>
`;
